src/lib/tls/tls_client.cpp

```diff
@@ -48,9 +48,14 @@ Client::Client(Callbacks& callbacks,
       }
 
    if(effective_version == Protocol_Version::TLS_V13)
+      {
       m_impl = std::make_unique<Client_Impl_13>(
                   callbacks, session_manager, creds, policy,
                   rng, info, next_protocols);
+
+      if(m_impl->expects_downgrade())
+         { m_impl->set_io_buffer_size(io_buf_sz); }
```

Comment:
I don't understand the connection between a downgrade and the need to set the IO buffer size here

Reply:
TL;DR: it's merely some parameter bookkeeping.

The TLS 1.2 stack has a notion of the I/O buffer size which is set during initialization. When downgrading from TLS 1.3 to 1.2, the TLS 1.3 instantiation is retired and a TLS 1.2 stack is instantiated. Without this we wouldn't have a way to know the user's setting for the I/O buffer size. Hence, this setter just takes note of it in the `Downgrade_Information` object.

Reply:
Got it :+1: